public/javascripts/case-progress.js

```diff
@@ -45,14 +45,22 @@
   // Edit note
 
   const getNotEditHandler = (hearingNoteDisplayContainer, noteEditContainer, noteReadonlyText, noteEditText, editing) => {
+    const trackEvent = (name, properties) => {
+      if (window && window.appInsights && window.appInsights.trackEvent && name) {
+        window.appInsights.trackEvent({ name, properties })
+        window.appInsights.flush()
+      }
+    }
     const originalText = noteReadonlyText.innerText
     const handler = (event) => {
       event.preventDefault()
       if (editing) {
```

Comment:
one and the same really, but you could've added this `trackEvent` as a one liner.

```javascript
trackEvent(editing ?  'PiCEditNoteStarted' : 'PiCEditNoteCancel', { hearingId: noteEditText.dataset.hearingid, noteId: noteEditText.dataset.noteid })
```